include/deal.II/differentiation/sd/symengine_tensor_operations.h

```diff
@@ -635,6 +635,121 @@ namespace Differentiation
 
     //@}
 
+    /**
+     * @name Symbol substitution and evaluation
+     */
+    //@{
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * tensor of symbolic expressions.
+     * The symbols in the @p expression_tensor that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * For more information regarding the performance of symbolic substitution,
+     * and the outcome of evaluation using a substitution map with cyclic
+     * dependencies, see the @ref substitute(const Expression &,
+     * const types::substitution_map &) function.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     */
+    template <int rank, int dim>
+    Tensor<rank, dim, Expression>
+    substitute(const Tensor<rank, dim, Expression> &expression_tensor,
+               const types::substitution_map &      substitution_map);
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * symmetric tensor of symbolic expressions.
+     * The symbols in the @p expression_tensor that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * For more information regarding the performance of symbolic substitution,
+     * and the outcome of evaluation using a substitution map with cyclic
+     * dependencies, see the @ref substitute(const Expression &,
+     * const types::substitution_map &) function.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     */
+    template <int rank, int dim>
+    SymmetricTensor<rank, dim, Expression>
+    substitute(const SymmetricTensor<rank, dim, Expression> &expression_tensor,
+               const types::substitution_map &               substitution_map);
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * tensor of symbolic expressions, and immediately evaluate the tensorial
+     * result.
+     * The symbols in the @p expression_tensor that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function is used to give a set of symbolic variables
+     * a numeric interpretation, with the returned result being of the type
```

Comment:
```suggestion
     * a numeric interpretation with the returned result being of the type
```
The comma just feels weird to me here.

Reply:
Ok, that's just a matter of style. But I'll happily change it.

Reply:
I am not the native speaker here. :smiley:

Reply:
Its no hassle either way :-)